CIPs/CIP-79/cip-79.md

```diff
@@ -0,0 +1,270 @@
+
+```
+3idv1 = "did:3:<cid>"
+```
+
+#### Example
+
+```sh
+did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi
+```
+
+## CRUD Operation Definitions
+
+In this section the CRUD operations for a 3ID DID are defined.
+
+### Create (Register)
+
+A `3` DID is created by simply creating a Ceramic [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document. The [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document type takes a DID as the *controller*, and it's recommended that a `did:key` is used. The *controller* is the DID which is allowed to update the document. The *family* of the document is set to `3id`, and the *deterministic* flag is set to `true`.
+
+Now the content of the document should consist of a JSON object with one property *publicKeys*. The value of this property should be an object where the value is a [multicodec](https://github.com/multiformats/multicodec/) *base58btc* encoded public key, the key for any given key should be the last *15* characters of the encoded public key.
+
+The 3ID DID method supports any type of public key that can be encoded using multicodec which can easily be extended to support quantum resistant signature and encryption schemes in the future.
+
+#### Example
+
+Below you can see an example of how to create a 3ID using the Ceramic javascript api.
+
+```js
+const doc = await ceramic.createDocument({
+  metadata: {
+    controllers: ['did:key:zQ3shQNcackrTByiYaPGso1Nt7b6r1gSMg4XXBmavzvTMqX1h'], // secp256k1 did:key
+    family: '3id'
+  },
+  content: {
+    publicKeys: {
+      "XQCT2xJHsdY6iJH": "z6LSqKWh3XQ7AfsJuE2KR23cozEut8D5CXQCT2xJHsdY6iJH",  // secp256k1 public key
+      "yh27jTt7Ny2Pwdy": "zQ3shrMGEKAjUTMmvDkcZ7Y3x9XnVjTH3myh27jTt7Ny2Pwdy", // x25519 public key
+    }
+  },
+  deterministic: true
+})
+
+const didString = `did:3:${doc.id}`
+```
+
+### Read (Resolve)
+
+Resolving a 3ID is quite straight forward. It is done by taking the DocID from the method specific identifier, looking up the referred document using Ceramic, and converting the content of the Ceramic document into a DID document.
+
+#### Loading 3IDv1 document
+
+To load a 3IDv1 document simply take the DocID from the method specific identifier and load the document from ceramic.
+
+#### Loading 3IDv0 document
+
+To load a 3IDv0 document, first take the CID from the method specific identifier and load the corresponding object from the IPFS network. The resolved object will contain a `publicKey` property which has an array of public key objects (see [Appendix A](#appendix-a)).
+
+Now find the key with an id that ends in `#signingKey` and convert it to a multicodec encoded key (in compressed format), then use it as a `did:key` and look up the ceramic document which has this key as the *controller* and *family* set to `3id`, using the *deterministic* flag. Below you can see the code to do this with the javascript Ceramic api.
```

Comment:
wait, so there are actually two ceramic 3id documents that need to be looked up?  So you take the did specifier, get the docId from it, look it up on ceramic, then find the `signingKey` (how do you know which of the keys is the signing key?), then you look up *another* 3id doc based on the signing key?  Now the rest of the document says "Once we have the Ceramic document loaded" but now we have two ceramic documents in play, right?  Which document is being referred to below?

I feel like I must not be understanding this section right

Reply:
No 3IDv0 loads the genesis object from IPFS:
> To load a 3IDv0 document, first take the CID from the method specific identifier and load the corresponding object from the IPFS network.

For 3IDv1 you just load the docId in the method specific identifier.

Reply:
Ugh I must have somehow forgotten I was reading about V0 midway through reading this section.  This all makes sense now, sorry for the confusion